provisioning/service/src/Enrollments/EnrollmentGroup.cs

```diff
@@ -317,20 +317,25 @@ public Attestation Attestation
         [JsonProperty(PropertyName = "allocationPolicy", DefaultValueHandling = DefaultValueHandling.Ignore)]
         public AllocationPolicy? AllocationPolicy { get; set; }
 
-#pragma warning disable CA2227 // Collection properties should be read only. Will not change public API
-
         /// <summary>
         /// The list of names of IoT hubs the device(s) in this resource can be allocated to. Must be a subset of tenant level list of IoT hubs
         /// </summary>
         [JsonProperty(PropertyName = "iotHubs", DefaultValueHandling = DefaultValueHandling.Ignore)]
-        public ICollection<string> IotHubs { get; set; }
-
-#pragma warning restore CA2227 // Collection properties should be read only
+        public IList<string> IotHubs { get; private set; } = new List<string>();
 
         /// <summary>
         /// Custom allocation definition.
         /// </summary>
         [JsonProperty(PropertyName = "customAllocationDefinition", DefaultValueHandling = DefaultValueHandling.Ignore)]
         public CustomAllocationDefinition CustomAllocationDefinition { get; set; }
+
+        /// <summary>
+        /// For use in serialization.
+        /// </summary>
+        /// <seealso href="https://www.newtonsoft.com/json/help/html/ConditionalProperties.htm#ShouldSerialize"/>
+        public bool ShouldSerializeIotHubs()
```

Comment:
Unfortunately must be public, but tells Newtonsoft not to serialize with custom critieria, in this case if the collection is empty.